server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -337,6 +411,48 @@ private void addToMap(Map<String, Set<Entitlement>> map, Entitlement e) {
         }
     }
 
+    public Set<Consumer> getDistinctConsumers(
+            List<Entitlement> entsToRevoke) {
+        Set<Consumer> result = new HashSet<Consumer>();
+        for (Entitlement ent : entsToRevoke) {
+            result.add(ent.getConsumer());
+        }
+        return result;
+    }
+    @SuppressWarnings("unchecked")
+    public ProductEntitlements getOverlappingForModifying(List<Entitlement> e) {
+        Date earliestStartDate = findEarliestStartDate(e);
+        Date latestEndDate = findLatestEndDate(e);
+        Set<Consumer> consumers = getDistinctConsumers(e);
+
+        List<Entitlement> overlapEnts = createModifiesDateFilteringCriteria(
+                consumers, earliestStartDate, latestEndDate)
+                .list();
+
+        return new ProductEntitlements(overlapEnts);
+    }
+
+    private Date findLatestEndDate(List<Entitlement> entitlements) {
+        Date max = null;
+        for (Entitlement e : entitlements) {
+            if (max == null || max.before(e.getEndDate())) {
```

Comment:
This Question is to both you and me ( cause I implemented the same thing in #1073 ) So I think we should ask @mstead .
@mstead by any chance do entitlements have null end dates that imply never ending entitlements? I ask cause I know our upstream source of information did do that in a lot of places.